todo_app/app.py

```diff
@@ -17,41 +21,109 @@ def del_DB(db):
 def create_app(db = None):
     app = Flask(__name__)
     app.config.from_object('todo_app.flask_config.Config')
+    app.config['LOGIN_DISABLED'] = os.getenv('LOGIN_DISABLED')
 
     if db == None:
         db = get_DB()
 
-    @app.route('/', methods=["GET"])
+    #login manager
+    login_manager = LoginManager()
+
+    client = WebApplicationClient(os.getenv('GITHUB_CLIENTID'))
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        authorization_endpoint = 'https://github.com/login/oauth/authorize'
+        request_uri = client.prepare_request_uri(
+            authorization_endpoint
+        )
+        return redirect(request_uri)
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        if user_id in users:
+            return users[user_id] 
+        else:
+            return None
+
+
+    login_manager.init_app(app)
+
+    @app.route('/users/callback', methods=["GET"])
+    def callback_func():
+        token_endpoint='https://github.com/login/oauth/access_token'
+        token_url, headers, body = client.prepare_token_request(
+            token_endpoint,
+            authorization_response=request.url,
+            redirect_url=request.base_url,
+            client_secret=os.getenv('GITHUB_CLIENTSECRET')
+        )
+        token_response = requests.post(
+            token_url,
+            headers=headers,
+            data=body,
+            auth=(os.getenv('GITHUB_CLIENTID'), os.getenv('GITHUB_CLIENTSECRET'))
```

Comment:
You should find you can remove these auth variables, as the `client_secret` is prepared as part of the body in the prepare_token_request method, and the client id is automatically included (since the client was set up with that originally)